player/src/main/java/com/focusit/jsflight/player/webdriver/SeleniumDriver.java

```diff
@@ -203,15 +239,35 @@ public void openEventUrl(WebDriver wd, JSONObject event, int pageTimeoutMs, Stri
             wd.get(event_url);
 
             waitUiShow(uiShownScript, wd);
-            waitPageReady(wd, event, pageTimeoutMs, checkPageJs);
+            waitPageReady(wd, event);
             updateLastUrl(event, event_url);
         }
     }
 
-    public void processKeyboardEvent(WebDriver wd, JSONObject event, WebElement element, boolean useRandomChars)
-            throws UnsupportedEncodingException
+    public void processKeyboardEvent(WebDriver wd, JSONObject event) throws UnsupportedEncodingException
     {
+        WebElement element = findTargetWebElement(wd, event, scenario.getTargetForEvent(event));
+        if (element.equals(NO_OP_ELEMENT))
+        {
+            LOG.warn("Non operational element returned. Aborting event processing. Target xpath {}",
```

Comment:
please add eventId or something alike to make it easier to identify a broken event